src/main/webapp/mapScript.js

```diff
@@ -132,32 +87,120 @@ function addNewTableItem(name, id) {
   tabel.appendChild(newPlace);
 }
 
-async function deletePlace(contentId) {
-  let tableItem = document.getElementById(contentId);
+function deletePlace(contentId) {
+  let tableItem = document.getElementById("place" + contentId);
   tableItem.parentNode.removeChild(tableItem);
   console.log("Remove place from the table");
 
   for (var i = 0; i < markersArray.length; i++) {
-    if (markersArray[i].id == contentId) {
+    if (markersArray[i].index == contentId) {
       markersArray[i].marker.setMap(null);
       markersArray.splice(i, 1);
       console.log("remove marker from the array and the map");
-      break;
     }
   }
+}
 
-  let options = {
-    method: "POST",
+function showSettings(contentId) {
+  var settings = document.getElementsByClassName("marker-setting")[0];
+  document.getElementById("submit-button").onclick = function () {
+    updateMarkerSettings(contentId);
   };
 
-  let URL = "/delete_marker?contentId=" + contentId;
-  await fetch(URL, options)
-    .then((response) => response.json())
-    .then((deletionResult) => {
-      if (deletionResult) {
-        console.log("Successful deletion");
-      } else {
-        console.log("Unsuccessful deletion");
-      }
-    });
+  // Show popup.
+  // setting.style.visibility = "visible";
+  settings.classList.toggle("show");
+  document.getElementById("marker-name").value = markersArray[contentId].name;
+  document.getElementById("visit-hour").value =
+    markersArray[contentId].visitHour;
+  document.getElementById("visit-minute").value =
+    markersArray[contentId].visitMinute;
+  document.getElementById("leave-hour").value =
+    markersArray[contentId].leaveHour;
+  document.getElementById("leave-minute").value =
+    markersArray[contentId].leaveMinute;
+}
+
+function updateMarkerSettings(contentId) {
+  let markerName = document.getElementById("marker-name").value,
+    visitHour = document.getElementById("visit-hour").value,
+    visitMinute = document.getElementById("visit-minute").value,
+    leaveHour = document.getElementById("leave-hour").value,
+    leaveMinute = document.getElementById("leave-minute").value;
+
+  markersArray[contentId].name = markerName;
+  markersArray[contentId].visitHour = visitHour;
+  markersArray[contentId].visitMinute = visitMinute;
+  markersArray[contentId].leaveHour = leaveHour;
+  markersArray[contentId].leaveMinute = leaveMinute;
+
+  let tableItemElemens = document.getElementById("place" + contentId)
+    .childNodes;
+  tableItemElemens[1].innerHTML = markerName;
+}
+
+async function createRoute() {
+  var routeName = document.getElementById("route-name").value;
+  if (routeName == "") {
+    alert("Please add a name to your new route!");
+  } else {
+    var markersIds = [];
+    console.log(markersArray);
+
+    for (var i = 0; i < markersArray.length; i++) {
```

Comment:
This approach of making two calls to the server for every marker is very inefficient. It would be better to send all the data to the server in one big JSON object and then do all the saves as part of a single POST request.

Reply:
Done